tests/chainer_tests/functions_tests/connection_tests/test_linear.py

```diff
@@ -41,138 +33,68 @@ def _to_noncontiguous(arrays):
     + [
         {'use_chainerx': True, 'chainerx_device': 'native:0'},
         {'use_chainerx': True, 'chainerx_device': 'cuda:0'},
-    ])
-class TestNonparameterizedLinear(unittest.TestCase):
+    ]
+)
+class TestNonparameterizedLinear(testing.FunctionTestCase):
 
     def setUp(self):
+        self.check_backward_options = {'atol': 1e-2, 'rtol': 1e-2}
+        self.check_double_backward_options = {'atol': 1e-2, 'rtol': 1e-2}
+        if self.x_dtype == numpy.float16:
+            self.check_forward_options = {'atol': 1e-3, 'rtol': 1e-2}
         self.n_batch_axes = self.x_shape['n_batch_axes']
+
+    def before_test(self, test_name):
+        if test_name == 'test_forward':
```

Comment:
Can you add a TODO comment, that this is temporary due to a known issue in the reduction of ChainerX, and that it should be removed?.